pennylane/math/sparse_matrix.py

```diff
@@ -0,0 +1,258 @@
+    (4, 4)
+    """
+
+    def __init__(self, arg):
+
+        if isinstance(arg, tuple):
+            self._data = {}
+            self._shape = arg
+            if len(self._shape) != 2:
+                raise ValueError(f"Expected a 2-dimensional tensor; got shape {self._shape}.")
+
+        else:
+            self._shape = qml.math.shape(arg)
+            if len(self._shape) != 2:
+                raise ValueError(f"Expected a 2-dimensional tensor; got shape {self._shape}.")
+
+            self._data = {}
+            for i in range(self._shape[0]):
+                for j in range(self._shape[1]):
+                    entry = arg[i, j]
+                    # easiest way to check that entry is not zero
+                    if qml.math.count_nonzero(entry) > 0:
+                        self._data[(i, j)] = entry
+
+    @property
+    def data(self):
+        """Return the data of the representation.
+
+        Returns:
+            dict
+        """
+        return self._data
+
+    @property
+    def shape(self):
+        """Return the shape of the matrix.
+
+        Returns:
+            tuple
+        """
+        return self._shape
+
+    @property
+    def row(self):
+        """Return a list of the first indices of nonzero entries.
+
+        Returns:
+            list[int]: list of row indices
+        """
+        return [idx[0] for idx in self.data]
+
+    @property
+    def col(self):
+        """Return a list of the second indices of nonzero entries.
+
+        Returns:
+            list[int]: list of column indices
+        """
+        return [idx[1] for idx in self.data]
+
+    @property
+    def nnz(self):
+        """Number of nonzero entries.
+
+        Returns:
+            int: number of elements
+        """
+        return len(self._data)
+
+    def __eq__(self, other):
+        """Comparison to another ``SparseMatrix`` object.
+        This method compares the data dictionaries.
+
+        Args:
+            other (SparseMatrix): sparse matrix to compare to
+
+        Returns:
+            bool
+        """
+        if not isinstance(other, type(self)):
+            return False
+        return self.data == other.data and self.shape == other.shape
+
+    def __repr__(self):
+        """Representation of this object.
+
+        Return:
+            str
+        """
+        return f"<SparseMatrix: entries={self.nnz}, shape={self.shape}>"
+
+    def __add__(self, other):
+        """Addition of another ``SparseMatrix`` object.
+
+        Args:
+            other (SparseMatrix): sparse matrix to add
+
+        Returns:
+            .SparseMatrix: sum of sparse matrices
+        """
+        if not isinstance(other, type(self)):
+            raise ValueError(f"Cannot add SparseMatrix and {type(other)}.")
+        if self.shape != other.shape:
+            raise ValueError(
+                f"Cannot add SparseMatrix object of different shape, got {other.shape}"
+            )
+
+        new = deepcopy(self)
```

Comment:
I have a feeling Torch tensors don't like to be copied 🤔 Did this show up during tests?

Reply:
No, strangely enough not!